File: backend-dotnet/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ClothingStore.Models
{
    [Table("orders")]
    public class Order
    {
        [Column("id")]
        public long Id { get; set; }
        
        [Column("user_id")]
        [Required]
        public long UserId { get; set; }
        
        [Column("order_date")]
        public DateTime OrderDate { get; set; }
        
        [Column("total_amount")]
        public decimal TotalAmount { get; set; }
        
        [Column("status")]
        public string Status { get; set; } = "PLACED";
        
        [Column("shipping_address")]
        public string? ShippingAddress { get; set; }
        
        [Column("city")]
        public string? City { get; set; }
        
        [Column("state")]
        public string? State { get; set; }
        
        [Column("pincode")]
        public string? Pincode { get; set; }
        
        [Column("phone")]
        public string? Phone { get; set; }
        
        [Column("payment_status")]
        public string? PaymentStatus { get; set; }
        
        [Column("total_items")]
        public int TotalItems { get; set; }
        
        [Column("order_id")]
        public string OrderId { get; set; } = string.Empty;
        
        [Column("delivery_date")]
        public DateTime? DeliveryDate { get; set; }
        
        [Column("status_reason")]
        public string? StatusReason { get; set; }
        
        [Column("cancel_date")]
        public DateTime? CancelDate { get; set; }
        
        [Column("coupon_code")]
        public string? CouponCode { get; set; }
        
        [Column("discount_amount")]
        public decimal? DiscountAmount { get; set; }
        
        [Column("message")]
        public string? Message { get; set; }
        
        [Column("payment_type")]
        public string? PaymentType { get; set; }
    }
}

File: genaibackend/src/main/java/com/clothingstore/entity/Order.java
package com.clothingstore.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "order_id", unique = true, nullable = false, length = 8)
    private String orderId;
    
    @Column(name = "user_id", nullable = false)
    private Long userId;
    
    @Column(nullable = false)
    private LocalDateTime orderDate;
    
    private LocalDateTime cancelDate;
    
    private LocalDateTime deliveryDate;
    
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;
    
    @Column(name = "discount_amount", precision = 10, scale = 2)
    private BigDecimal discountAmount = BigDecimal.ZERO;
    
    @Column(name = "coupon_code")
    private String couponCode;
    
    @Column(name = "total_items", nullable = false)
    private Integer totalItems = 0;
    
    @Column(name = "message", length = 500)
    private String message;
    
    @Enumerated(EnumType.STRING)
    private OrderStatus status;
    
    @Enumerated(EnumType.STRING)
    private PaymentStatus paymentStatus;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "payment_type")
    private PaymentType paymentType;
    
    private String shippingAddress;
    private String city;
    private String state;
    private String pincode;
    
    @Column(length = 13)
    private String phone;
    
    private String statusReason;
    
    public enum OrderStatus {
        PLACED, CONFIRMED, SHIPPED, OUT_FOR_DELIVERY, DELIVERED, 
        CANCELLED, REFUNDED, EXCHANGED, RETURNED,
        REQUEST_CANCEL, REQUEST_EXCHANGE, REQUEST_REPLACE, REQUEST_RETURN
    }
    
    public enum PaymentStatus {
        PENDING, PROCESSING, COMPLETED, FAILED, REFUNDED, PARTIALLY_REFUNDED
    }
    
    public enum PaymentType {
        ONLINE, COD
    }
    
    // Constructors
    public Order() {
        this.orderDate = LocalDateTime.now();
        this.status = OrderStatus.PLACED;
        this.paymentStatus = PaymentStatus.PENDING;
        this.orderId = generateOrderId();
    }
    
    public Order(Long userId, BigDecimal totalAmount) {
        this();
        this.userId = userId;
        this.totalAmount = totalAmount;
    }
    
    private String generateOrderId() {
        return String.valueOf((int)(Math.random() * 90000000) + 10000000);
    }
    
    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public String getOrderId() { return orderId; }
    public void setOrderId(String orderId) { this.orderId = orderId; }
    
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    
    public LocalDateTime getOrderDate() { return orderDate; }
    public void setOrderDate(LocalDateTime orderDate) { this.orderDate = orderDate; }
    
    public LocalDateTime getCancelDate() { return cancelDate; }
    public void setCancelDate(LocalDateTime cancelDate) { this.cancelDate = cancelDate; }
    
    public LocalDateTime getDeliveryDate() { return deliveryDate; }
    public void setDeliveryDate(LocalDateTime deliveryDate) { this.deliveryDate = deliveryDate; }
    
    public BigDecimal getTotalAmount() { return totalAmount; }
    public void setTotalAmount(BigDecimal totalAmount) { this.totalAmount = totalAmount; }
    
    public BigDecimal getDiscountAmount() { return discountAmount; }
    public void setDiscountAmount(BigDecimal discountAmount) { this.discountAmount = discountAmount; }
    
    public String getCouponCode() { return couponCode; }
    public void setCouponCode(String couponCode) { this.couponCode = couponCode; }
    
    public Integer getTotalItems() { return totalItems; }
    public void setTotalItems(Integer totalItems) { this.totalItems = totalItems; }
    
    public String getMessage() { return message; }
    public void setMessage(String message) { this.message = message; }
    
    public OrderStatus getStatus() { return status; }
    public void setStatus(OrderStatus status) { this.status = status; }
    
    public PaymentStatus getPaymentStatus() { return paymentStatus; }
    public void setPaymentStatus(PaymentStatus paymentStatus) { this.paymentStatus = paymentStatus; }
    
    public String getShippingAddress() { return shippingAddress; }
    public void setShippingAddress(String shippingAddress) { this.shippingAddress = shippingAddress; }
    
    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }
    
    public String getState() { return state; }
    public void setState(String state) { this.state = state; }
    
    public String getPincode() { return pincode; }
    public void setPincode(String pincode) { this.pincode = pincode; }
    
    public String getPhone() { return phone; }
    public void setPhone(String phone) { 
        if (phone != null && !phone.startsWith("+91")) {
            this.phone = "+91" + phone;
        } else {
            this.phone = phone;
        }
    }
    
    public String getStatusReason() { return statusReason; }
    public void setStatusReason(String statusReason) { this.statusReason = statusReason; }
    
    public PaymentType getPaymentType() { return paymentType; }
    public void setPaymentType(PaymentType paymentType) { this.paymentType = paymentType; }
}